examples/dinput8_hook_d3d12/src/lib.rs

```diff
@@ -0,0 +1,114 @@
+#![feature(once_cell)]
+#![allow(non_snake_case, unused_variables, unreachable_patterns)]
+
+use std::ffi::c_void;
+use std::sync::LazyLock;
+use std::thread;
+
+use hudhook::hooks::dx12::ImguiDx12Hooks;
+use hudhook::hooks::{self, ImguiRenderLoop, ImguiRenderLoopFlags};
+use hudhook::log::trace;
+use hudhook::reexports::DLL_PROCESS_ATTACH;
+use hudhook::renderers::imgui_dx12::imgui;
+use imgui::Condition;
+use simplelog::{Color, ColorChoice, ConfigBuilder, Level, LevelFilter, TermLogger, TerminalMode};
+use windows::core::{GUID, HRESULT, PCSTR};
+use windows::s;
+use windows::Win32::Foundation::{BOOL, HINSTANCE};
+use windows::Win32::System::LibraryLoader::{GetProcAddress, LoadLibraryA};
+use windows::Win32::UI::WindowsAndMessaging::{LoadCursorW, SetCursor, IDC_ARROW};
+struct Dx12HookExample;
+
+impl Dx12HookExample {
+    fn new() -> Self {
+        hudhook::utils::alloc_console();
+        let config = ConfigBuilder::new()
+            .set_level_color(Level::Error, Some(Color::Magenta))
+            .set_level_color(Level::Trace, Some(Color::Green))
+            .set_target_level(LevelFilter::Error)
+            .build();
+        TermLogger::init(LevelFilter::Error, config, TerminalMode::Mixed, ColorChoice::Auto).ok();
+
+        Dx12HookExample
+    }
+}
+
+impl ImguiRenderLoop for Dx12HookExample {
+    fn should_block_messages(&self, _io: &mut imgui::Io) -> bool {
```

Comment:
When a parameter name is actually used, I'd prefer the leading `_` to be removed.
Also, the `Io` argument should _not_ be `&mut` as this method is meant to be a predicate -- see below for a more complete explanation.